src/ibc/types.rs

```diff
@@ -0,0 +1,252 @@
+/// IBC connection
+pub struct Connection {
+    /// Connection id on the counterparty chain
+    pub counterparty_connection_id: Option<String>,
+    /// Chain id of the counterparty chain
+    pub counterparty_chain_id: String,
+}
+
+#[cosmwasm_schema::cw_serde]
+#[derive(Default)]
+/// IBC Port Info
+pub struct PortInfo {
+    /// Channel id of the next opened channel
+    pub next_channel_id: u64,
+}
+
+#[cosmwasm_schema::cw_serde]
+pub struct ChannelHandshakeInfo {
+    pub connection_id: String,
+    pub port: MockIbcPort,
+    pub local_endpoint: IbcEndpoint,
+    pub remote_endpoint: IbcEndpoint,
+    pub state: ChannelHandshakeState,
+    pub order: IbcOrder,
+    pub version: String,
+}
+
+#[cosmwasm_schema::cw_serde]
+pub enum ChannelHandshakeState {
+    Init,
+    Try,
+    Ack,
+    Confirm,
+}
+
+#[cosmwasm_schema::cw_serde]
+pub struct ChannelInfo {
+    pub next_packet_id: u64,
+    pub last_packet_relayed: u64,
+
+    pub info: IbcChannel,
+
+    pub open: bool,
+}
+
+#[cosmwasm_schema::cw_serde]
+pub enum MockIbcPort {
+    Wasm(String), // A wasm port is simply a wasm contract address
+    Bank,         // The bank port simply talks to the bank module
+    Staking,      // The staking port simply talks to the staking module
+}
+
+impl From<MockIbcPort> for IbcModule {
+    fn from(port: MockIbcPort) -> IbcModule {
+        match port {
+            MockIbcPort::Bank => IbcModule::Bank,
+            MockIbcPort::Staking => IbcModule::Staking,
+            MockIbcPort::Wasm(contract) => IbcModule::Wasm(Addr::unchecked(contract)),
+        }
+    }
+}
+
+pub const BANK_MODULE_PORT: &str = "transfer";
+
+impl Display for MockIbcPort {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            MockIbcPort::Wasm(c) => write!(f, "wasm.{}", c),
+            MockIbcPort::Bank => write!(f, "{BANK_MODULE_PORT}"),
+            MockIbcPort::Staking => panic!("No ibc port for the staking module"),
+        }
+    }
+}
+
+impl FromStr for MockIbcPort {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // For the bank module
+        if s.eq(BANK_MODULE_PORT) {
+            return Ok(MockIbcPort::Bank);
+        }
+
+        // For the wasm module
+        let wasm = s.split('.').collect::<Vec<_>>();
+        if wasm.len() == 2 && wasm[0] == "wasm" {
+            return Ok(MockIbcPort::Wasm(wasm[1].to_string()));
+        }
+        // Error
+        bail!(
+            "The ibc port {} can't be linked to an mock ibc implementation",
+            s
+        )
+    }
+}
+
+#[cosmwasm_schema::cw_serde]
+pub struct IbcPacketData {
+    pub ack: Option<Binary>,
+    /// This also tells us whether this packet was already sent on the other chain or not
+    pub src_port_id: String,
+    pub src_channel_id: String,
+    pub dst_port_id: String,
+    pub dst_channel_id: String,
+    pub sequence: u64,
+    pub data: Binary,
+    pub timeout: IbcTimeout,
```

Comment:
Couldn't this type just have an `cosmwasm_std::IbcPacket` field?

Reply:
Using that now